src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeNameBuilder.cs

```diff
@@ -0,0 +1,325 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class TypeNameBuilder
```

Comment:
This file is line-to-line identical to https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/TypeNameBuilder.cs except for deleted comment and unused method.

Would it be better to just include the CoreLib in the .csproj file instead of copying it the file over?

Reply:
> This file is line-to-line identical to https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/TypeNameBuilder.cs except for deleted comment and unused method.

Right, and I mentioned that in the description.

> Would it be better to just include the CoreLib in the .csproj file instead of copying it the file over?

Was not aware of that was an option, I will try that out with my next PR which probably will not happen in .NET 8.0, would you like to update that within .NET 8?

Reply:
We have number of places where we borrow CoreLib code to avoid copying it around. Look for `<Compile Include="$(CoreLibSharedDir)System...`. This pattern is typically used for down-level configs, I do not see a problem with doing the same here.

This does not need to be in .NET 8. However, I do prefer smaller more frequent PRs that do just one thing - they are much easier to review and merge.